libtiledbsoma/src/soma/managed_query.cc

```diff
@@ -1218,23 +1373,29 @@ ManagedQuery::_extend_and_evolve_schema_with_details(
     ArrowSchema* value_schema,
     ArrowArray* value_array,
     std::string column_name,
+    bool deduplicate,
     Enumeration enmr,
     ArraySchemaEvolution& se) {
-    // We need to check if we are writing any new enumeration values. If so,
-    // extend and evolve the schema. If not, just set the write buffers to the
-    // dictionary's indexes as-is
+    if (value_array->n_buffers != 2) {
+        throw std::invalid_argument(fmt::format(
```

Comment:
this error is going to confuse most users (I think).  Are you really just prohibiting dictionary arrays?

Reply:
The check is important. I found this the hard way when, in development, I was passing a Python categorical array rather than the values array.

The check needs to remain here, at this level. This is a fallback, ensuring that before we deference `buffers[1]` or `buffers[2]` that they actually exist. I will more clearly word this to indicate that fact. The context I didn't make sufficiently clear is that it is an internal coding error if this check ever fails.

Reply:
Also: https://github.com/single-cell-data/TileDB-SOMA/pull/3815/commits/f215424fcbcbfdfa7c3c50c171a8f5106e43e5d0

Reply:
pushed